bugbug/tools/code_review.py

```diff
@@ -1116,15 +1119,23 @@ def find_similar_patch_comments(self, patch: Patch, limit: int):
 
         patch_set = PatchSet.from_string(patch.raw_diff)
 
-        yielded = 0
+        # We want to avoid returning the same comment multiple times. Thus, if
+        # a comment matches multiple hunks, we will only consider it once.
+        max_score_per_comment: dict = {}
         for patched_file in patch_set:
             if not patched_file.is_modified_file:
                 continue
 
             for hunk in patched_file:
                 for result in self.find_similar_hunk_comments(hunk):
-                    yield result
+                    if result is not None and (
+                        result.id not in max_score_per_comment
+                        or result.score > max_score_per_comment[result.id].score
+                    ):
+                        max_score_per_comment[result.id] = result
+
+        results_with_score = sorted(
+            (result.score, result) for result in max_score_per_comment.values()
+        )
```

Comment:
We could rely on the `PayloadScore` dataclass to be sortable and then do the following:
```suggestion
        sorted_results = sorted(
            result for result in max_score_per_comment.values()
        )
```

To make the `PayloadScore` dataclass sortable, we need to pass `order=True` to `dataclass()` and re-order the fields to have `score` as the first one.